ckan/tests/controllers/test_group.py

```diff
@@ -60,63 +59,77 @@ def test_invalid_sort_param_does_not_crash(self, app):
             app.get(url=group_url)
 
 
-@pytest.mark.usefixtures("non_clean_db", "with_request_context")
+@pytest.fixture
+def sysadmin():
+    user = factories.Sysadmin(password="correct123")
+    user_token = factories.APIToken(user=user["name"])
+    env = {"Authorization": user_token["token"]}
+    data = {"env": env, "user_dict": user}
+    return data
+
+
+@pytest.fixture
+def user():
+    user = factories.User(password="correct123")
+    user_token = factories.APIToken(user=user["name"])
+    env = {"Authorization": user_token["token"]}
+    data = {"env": env, "user_dict": user}
+    return data
```

Comment:
I like the idea of the fixture but I'm not quite convinced on returning the `env` and the `user_dict` since it's a little bit confusing: Why a user dict has an `env` attribute?

I would refactor it to be:
```python
def user():
    user = factories.User(password="correct123")
    user["token"] = factories.APIToken(user=user["name"])
    return user
```

And later in the tests:
```python
    def test_name_required(self, app, user):
        env = {"Authorization": user["token"]}
        url = url_for("group.new")
        response = app.post(url=url, extra_environ=env, data={"save": ""})

        assert "Name: Missing value" in response
```


I think it's:
- more legible, 
- the fixture is isolated to be a user attribute
- We have the same amount of lines

Reply:
In fact, the `User` model have an attribute of API Key. Can we return a `user_dict` with the attribute `apikey` directly from the factory?

That way we would reduce the overhead of calling `user["token"] = factories.APIToken(user=user["name"])`.

Reply:
We cannot use apikey in 2.10 as we removed this line of code: 
https://github.com/ckan/ckan/blob/ca537ad25e13320b6fbf91f41031eadc30500f12/ckan/views/__init__.py#L231
on #6352